quic/client/handshake/FizzClientContext.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+#pragma once
+
+#include <quic/client/handshake/ClientHandshakeFactory.h>
+
+namespace quic {
+
+class FizzClientHandshake;
+
+class FizzClientContext
```

Comment:
Feedback from @siyengar, this name should probably be `FizzClientHandshakeFactory` to show that it's a factory.

Reply:
I named it this way because I want to subsequently move some fizz specific stuff in the context, for instance for connect.

Reply:
i think we should still name this FizzClientHandshakeFactory because the FizzClientContext is actually a specific object class name in fizz, so it will get very confusing otherwise. I think even if we move the context in here, it's useful to have a different name

Reply:
It end up being more than a factory, and effectively encapsulate `fizz::client::FizzClientContext` , so that seemed like an apt name.

Reply:
ya however i think that the name is too confusing, i would just prefer having the name be a factory even if it encaps multiple things

Reply:
FizzClientHandshakeContext ?